arrow-ord/src/cmp.rs

```diff
@@ -210,17 +210,19 @@ fn compare_op(op: Op, lhs: &dyn Datum, rhs: &dyn Datum) -> Result<BooleanArray,
         assert_eq!(l.num_columns(), r.num_columns());
         match op {
             Op::Equal => {
-                let mut res = BooleanArray::from(vec![true; len]);
-                for i in 0..l.num_columns() {
-                    let col_l = l.column(i);
-                    let col_r = r.column(i);
-                    let eq_rows = compare_op(op, col_l, col_r)?;
-
-                    let nulls = NullBuffer::union(res.nulls(), eq_rows.nulls());
-                    let vals = res.values() & eq_rows.values();
-                    res = BooleanArray::new(vals, nulls);
-                }
-                return Ok(res);
+                return (0..l.num_columns()).fold(
+                    Ok(BooleanArray::from(vec![true; len])),
```

Comment:
```suggestion
                    Ok(BooleanArray::new(BooleanBuffer::new_set(len), None)),
```

This will be significantly faster